src/compat/service.rs

```diff
@@ -222,19 +240,31 @@ impl Service {
         blacklist: HashSet<PaAddr>,
         crust_user: CrustUser,
     ) -> Result<(), CrustError> {
+        #[cfg(test)]
+        let disable_peer_heartbeats = self.disable_peer_heartbeats;
         self.event_loop
             .send(Box::new(move |state: &mut ServiceState| {
                 let (drop_tx, drop_rx) = future_utils::drop_notify();
                 let use_service_discovery = state.service_discovery_enabled;
                 state.bootstrap_connect = Some(drop_tx);
                 let cm = state.cm.clone();
                 let event_tx = state.event_tx.clone();
+                #[allow(unused_mut)]
                 let f = {
                     let handle = state.service.handle().clone();
                     state
                         .service
                         .bootstrap(blacklist, use_service_discovery, crust_user)
                         .map_err(|e| error!("bootstrap failed: {}", e))
+                        .map(move |mut peer| {
+                            #[cfg(test)]
```

Comment:
Not sure we can, but would see if we can avoid inlining selective compilation this way. Maybe we can structure it a little better to not have these sort of code at all ?